src/components/JSONInspector/JSONRenderer/useFakeScroll.ts

```diff
@@ -1,41 +1,66 @@
 import {
   type Dispatch,
   type SetStateAction,
+  useCallback,
   useLayoutEffect,
   useRef,
   useState,
 } from 'react'
 
+/**
+ * Controll scrollable container using `setCursor` based on computed `lineHeight`.
+ * @typeparam M - Type of the dummy element to measure line height
+ * @typeparam S - Type of the scrollable container element
+ */
 export function useFakeScroll<M extends HTMLElement, S extends HTMLElement>() {
   const [cursor, setCursor] = useState(0)
   const { lineHeight, measureRef } = useLineHeightMeasure<M>()
   const scrollRef = useFakeScrollPosition<S>(lineHeight, cursor, setCursor)
 
   return {
+    /**
+     * real number representation of current scroll position
+     * which of unit is 'line'
+     */
     cursor,
+    /**
+     * Set scroll position in line unit
+     */
     setCursor,
+    /**
+     * Reference to the dummy element to measure line height.
+     * It's good to add aria-hidden attribute to such dummy element.
+     */
     measureRef,
+    /**
+     * Reference to the scrollable container element.
+     */
     scrollRef,
+    /**
+     * Computed line height of the dummy element in pixel.
+     * If dummy element is demounted or not mounted, this value will be undefined.
+     */
     lineHeight,
   }
 }
 
+/**
+ * Retrieve line height of the element through `measureRef` initialization.
+ */
 function useLineHeightMeasure<E extends HTMLElement>() {
   const [lineHeight, setLineHeight] = useState<number>()
-  const measureRef = useRef<E>(null)
 
-  useLayoutEffect(() => {
-    function computeHeight() {
-      if (!measureRef.current) return
-
-      const lineHeight = measureRef.current.clientHeight
+  const handler = useRef<() => void>(() => {})
 
-      setLineHeight(lineHeight)
+  const measureRef = useCallback((element: E | null) => {
```

Comment:
Use deterministic DOM ref initialization using function ref, and clean up resize event listener properly